### ISSUE TITLE
Show publishing status in page in form editor, not in messages

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-flow#2488]
# Context

As @kymni well noted in this issue https://github.com/akvo/akvo-flow/issues/1403 publishing of a form and the end result of publishing does not happen immediately: "The publish survey endpoint does not immediately return the result of the publishing request because checks have to be first be run on the back-end. The status therefore cannot be changed without first knowing if the publishing request was successful or not." (https://github.com/akvo/akvo-flow/issues/1403#issuecomment-359427399) 

## User steps 
Petra is working on a survey form. She is adding in questions and making changes to the form structure. Once she is done with the form and it is ready for data collection she needs to publish it. By publishing she indicates that the form is no ready in draft stage but is now ready for enumerators to use. She hits on the publish button in the form and then gets a message telling her that she needs to go to the Messages tab to see that the publishing is completed. Now she is expected to leave the page. 

https://cloud.githubusercontent.com/assets/12456965/23306893/b1568134-faa5-11e6-9929-814e13a3c49c.gif

# Issues with this implementation 
- Once Petra hits the publish button she needs to go to the Messages tab to see the result --> this is unneeded extra navigation for her. 
- Once she hits publish, she should navigate away from the form and not make further changes to the form. However, sometimes she hits publish and remains on the form page and edits another question, or makes another change. --> Then the form version does not update properly and this causes issues with which change is published and which is not. 
- There is no visual indication of the form being published besides the disabled publish button. 

# Opportunities 

## Show publishing status in page, in form editor (not in messages tab) 
An idea is to show the status of publishing directly in the form window in a way that: 
- Petra cannot make any further edits till the publishing is done and successful 
- Petra sees that something is going on and needs to wait till it is completed 
- Petra can clearly see that a form is published or not https://github.com/akvo/akvo-flow/issues/1328

With this change we could then also remove the publish form message from the Messages tab

**Questions to resolve** 
- how long does it take to publish a form? 
- how long does it take to publish a form in low connectivity areas? 
- what currently makes the publishing not work? 

@Kiarii @muloem and @kymni Can you share your thoughts on this idea and details on the questions above? 

# Other related opportunities that have separate issues 

## Form integrity check

In the past we mentioned that we should validate that the form is "ready" for data collection with publishing. Many things we validate at the time of editing the form, like if a question has a name, or if an option question has options, if a cascade has a cascade, if dependency is invalidated after moving a questions around. If we want to add this to the publishing, we should handle it in a separate issue: https://github.com/akvo/akvo-flow/issues/1324

## Handling changes to cascades 
We also noted in a separate issue that if a change is made to a cascade that is added to a published form, the form should not be published. This is a complex issues.

**Comment by Kiarii** [comment]
me quick take: its going to be some improvement to show:
* Form status: published, draft 
* Distinguish between draft and published form update..
* Show if a published form has unsaved/unpublished edits - or prevent this from being made
* Status process in page - to remove the need to go back and forth between survey and msg tabs

**Comment by kymni** [comment]
We can do interval calls to the surveys endpoint to check if `status` property has changed to `PUBLISHED` similar to how we're checking for reports https://github.com/akvo/akvo-flow/blob/d1ba03912e57cb3a8303df09a3379b93e6206180/Dashboard/app/js/lib/controllers/reports-controllers.js#L53-L66 
This will have to change once the ember library is updated to accommodate object reloading

**Comment by janagombitova** [comment]
Still an issue and takes up a lot of support time. Needs love

**Comment by janagombitova** [comment]
Moving this issue to https://github.com/akvo/akvo-flow/issues/3795 so we have all the issues in one place to start with
